main/test/flix/Test.Exp.Lazy.flix

```diff
@@ -0,0 +1,95 @@
+namespace Test/Exp/Lazy {
+
+    @test
+    def testUnitLazy01(): Lazy[Unit] & Pure =
+        lazy ()
+
+    @test
+    def testBoolLazy01(): Lazy[Bool] & Pure =
+        lazy true
+
+    @test
+    def testCharLazy01(): Lazy[Char] & Pure =
+        lazy 'a'
+
+    @test
+    def testFloat32Lazy01(): Lazy[Float32] & Pure =
+        lazy 12.34f32
+
+    @test
+    def testFloat64Lazy01(): Lazy[Float64] & Pure =
+        lazy 12.34f64
+
+    @test
+    def testInt8Lazy01(): Lazy[Int8] & Pure =
+        lazy 12i8
+
+    @test
+    def testInt16Lazy01(): Lazy[Int16] & Pure =
+        lazy 12i16
+
+    @test
+    def testInt32Lazy01(): Lazy[Int32] & Pure =
+        lazy 12i32
+
+    @test
+    def testInt64Lazy01(): Lazy[Int64] & Pure =
+        lazy 12i64
+
+    @test
+    def testBigIntLazy01(): Lazy[BigInt] & Pure =
+        lazy 12ii
+
+    @test
+    def testStringLazy01(): Lazy[String] & Pure =
+        lazy "string"
+
+    @test
+    def testArrayOfIntLazy01(): Lazy[Array[Int]] & Impure =
+        let array = [1, 2];
+        lazy array
+
+    @test
+    def testArrayOfStringLazy01(): Lazy[Array[String]] & Impure =
+        let array = ["str", "str2"];
```

Comment:
Any need to use local variables?

Reply:
The expression given to lazy is required to be pure so array literals cant be given directly